x-pack/test/alerting_api_integration/common/plugins/actions_simulators/server/xsoar_simulation.ts

```diff
@@ -0,0 +1,421 @@
+  owner: '',
+  activated: '0001-01-01T00:00:00Z',
+  closeReason: '',
+  rawCloseReason: '',
+  closeNotes: '',
+  playbookId: '8db0105c-f674-4d83-8095-f95a9f61e77a',
+  dueDate: '2024-05-01T11:47:08.107988742Z',
+  reminder: '0001-01-01T00:00:00Z',
+  runStatus: '',
+  notifyTime: '0001-01-01T00:00:00Z',
+  phase: '',
+  rawPhase: '',
+  isPlayground: false,
+  rawJSON: '',
+  parent: '',
+  parentXDRIncident: '',
+  retained: false,
+  category: '',
+  rawCategory: '',
+  linkedIncidents: null,
+  linkedCount: 0,
+  droppedCount: 0,
+  sourceInstance: '',
+  sourceBrand: 'Manual',
+  canvases: null,
+  lastJobRunTime: '0001-01-01T00:00:00Z',
+  feedBased: false,
+  dbotMirrorId: '',
+  dbotMirrorInstance: '',
+  dbotMirrorDirection: '',
+  dbotDirtyFields: null,
+  dbotCurrentDirtyFields: null,
+  dbotMirrorTags: null,
+  dbotMirrorLastSync: '0001-01-01T00:00:00Z',
+  isDebug: false,
+};
+
+export const XSOARFailedResponse = {
+  id: 'incCreateErr',
+  status: 400,
+  title: 'Failed creating incident',
+  detail: 'Failed creating incident',
+  error: '',
+  encrypted: false,
+  multires: null,
+};
+
+export class XSOARSimulator extends Simulator {
+  private readonly returnError: boolean;
+
+  constructor({ returnError = false, proxy }: { returnError?: boolean; proxy?: ProxyArgs }) {
+    super(proxy);
+
+    this.returnError = returnError;
+  }
+
+  public async handler(
+    request: http.IncomingMessage,
+    response: http.ServerResponse,
+    data: Record<string, unknown>
+  ) {
+    if (this.returnError) {
+      return XSOARSimulator.sendErrorResponse(response);
+    }
+    return XSOARSimulator.sendResponse(request, response);
+  }
+
+  private static sendResponse(request: http.IncomingMessage, response: http.ServerResponse) {
+    response.setHeader('Content-Type', 'application/json');
+    let body;
+    if (request.url?.match('/incident')) {
+      response.statusCode = 201;
+      body = XSOARRunSuccessResponse;
+    } else if (request.url?.match('/playbook/search')) {
+      response.statusCode = 200;
+      body = XSOARPlaybooksResponse;
+    }
+    response.end(JSON.stringify(body, null, 4));
+  }
+
+  private static sendErrorResponse(response: http.ServerResponse) {
+    response.statusCode = 400;
+    response.setHeader('Content-Type', 'application/json');
+    response.end(JSON.stringify(XSOARFailedResponse, null, 4));
+  }
+}
+
+export function initPlugin(router: IRouter, path: string) {
+  router.post(
+    {
+      path: `${path}/playbook/search`,
+      options: {
+        authRequired: false,
+      },
+      validate: {},
+    },
+    async function (
+      context: RequestHandlerContext,
+      req: KibanaRequest<any, any, any, any>,
+      res: KibanaResponseFactory
+    ): Promise<IKibanaResponse<any>> {
+      return res.ok({ body: XSOARPlaybooksResponse });
+    }
+  );
+
+  router.post(
+    {
+      path: `${path}/incident`,
+      options: {
+        authRequired: false,
+      },
+      validate: {},
```

Comment:
There's a TS error here, let's add the following:
```suggestion
      validate: {},
      security: { authz: { enabled: false, reason: 'This route is opted out from authorization' } },
```